### PR TITLE
ipq806x: add support for Linksys e8350-v1

### PR DESCRIPTION
**Pull request by SergeyFilippov**
AC2400 Dual-Band Gigabit Wi-Fi Router base on ipq8064. 
https://www.linksys.com/support-product?sku=E8350

Specification:
```
 - Qualcomm dual-core IPQ8064 @ 1.4 GHz
 - 512 MB of RAM
 - 4 MB of SPI NOR MX25U3235F
 - 128 MB of NAND S34MS01G2
 - Qualcomm QCA9880 2.4GHz 802.11bgn
 - Quantenna QSR1000 5GHz 802.11ac (no support)
 - 4 x 10/100/1000 Mbit/s w/ vlan support Ethernet
 - Qualcomm Atheros QCA8337 switch
 - 1 x 3.0 + 1 x 2.0 (combo with eSata port)
 - 115200, 8N1 internal serial console
 - Power, Reset, WPS and WLAN buttons
 - Power, WPS and WLAN leds
 - 12 VDC, 3 A power
 ```

Installation:
The installation must be done using web interface of the router. To achive this new firmware-utils tool was added to set correct magic headers for the factory images.
https://github.com/openwrt/firmware-utils/pull/30

Installation from vendor firmware:
 1. Flash over the native Linksys WEB interface using factory image.

Installation using recovery mode:
 1. Power off the device and disconnect the WAN port. (Only LAN port to be connected)
 2. Press & hold the "Reset" button
 3. Power on the device & wait 10 seconds with pressed "Reset" button
 4. Set IP Internet Protocol on your PC from 192.168.1.0/24 network (Router is on IP 192.168.1.1)
 5. Open the Firmware Recovery page in your browser: http://192.168.1.1/index.shtml Firmware Recovery -> File Name -> Recovery & Reboot

The device page in inbox:
https://openwrt.org/inbox/toh/linksys/linksys_ea8350_1

### CODE REVIEW

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+};
+
+&tcsr {
+	qcom,usb-ctrl-select = <TCSR_USB_SELECT_USB3_DUAL>;
+	compatible = "qcom,tcsr";
+};
+
+&adm_dma {
+	status = "okay";
+};
```
Comment (Ansuel) [inline target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:428]: Nit, please add new line at end of the file.

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+	qcom,id = <2>;
+	
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&tcsr {
+	qcom,usb-ctrl-select = <TCSR_USB_SELECT_USB3_DUAL>;
```
Comment (Ansuel) [inline target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:422]: This is a good time to ask for this... Is it actually required? Does usb works without this?

**Comment by musashino205** [comment]
Please use `ipq806x: ` as a prefix in the commit title instead.

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts (hunk 1 above)
Comment (SergeyFilippov) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts]: should be fixed now

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts (hunk 2 above)
Comment (SergeyFilippov) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:422]: I've tried completely commenting it out, and the router stuck in boot (waited 5 minutes). After reintroducing it went back to a normal operation. So, for this model it seems to be required.

**Comment by SergeyFilippov** [comment]
> Please use `ipq806x: ` as a prefix in the commit title instead.

Done

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts (hunk 2 above)
Comment (Ansuel) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:422]: Mhhh no serial? Sorry if i'm asking this but I always wanted to upstream this tcsr thing and wa always confused by it.

Comment (SergeyFilippov) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:422]: Up until now there was no need in serial (thanks to other members who did research earlier).
But sure, I can try this out with serial connected. Will take some time tho (but most probably until the end of the week there will be some logs).

Comment (SergeyFilippov) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:422]: > Mhhh no serial? Sorry if i'm asking this but I always wanted to upstream this tcsr thing and wa always confused by it.

@Ansuel, actually I've managed to boot it, if completely commented out the whole section of tcsr.

```
[    6.087604] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    6.087654] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 1
[    6.092195] xhci-hcd xhci-hcd.0.auto: hcc params 0x0228f065 hci version 0x100 quirks 0x0000008002000010
[    6.099577] xhci-hcd xhci-hcd.0.auto: irq 47, io mem 0x10000000
[    6.109149] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    6.114892] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    6.120375] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    6.128566] hub 1-0:1.0: USB hub found
[    6.134603] hub 1-0:1.0: 1 port detected
[    6.138572] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    6.142865] hub 2-0:1.0: USB hub found
[    6.150429] hub 2-0:1.0: 1 port detected
[    6.155681] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    6.158054] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 3
[    6.163543] xhci-hcd xhci-hcd.1.auto: hcc params 0x0228f065 hci version 0x100 quirks 0x0000008002000010
[    6.170925] xhci-hcd xhci-hcd.1.auto: irq 48, io mem 0x11000000
[    6.180492] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    6.186244] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 4
[    6.191806] xhci-hcd xhci-hcd.1.auto: Host supports USB 3.0 SuperSpeed
[    6.199734] hub 3-0:1.0: USB hub found
[    6.205951] hub 3-0:1.0: 1 port detected
[    6.209908] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    6.214133] hub 4-0:1.0: USB hub found
[    6.221854] hub 4-0:1.0: 1 port detected
```
trying to attach a USB device fails to the first port
```
[  193.543764] usb usb4-port1: connect-debounce failed
```
works for a second one (e-sata):
```
[  246.511736] usb 1-1: new high-speed USB device number 2 using xhci-hcd
[  246.697096] usb-storage 1-1:1.0: USB Mass Storage device detected
[  246.697437] usb-storage 1-1:1.0: Quirks match for vid 174c pid 55aa: 400000
[  246.702390] scsi host0: usb-storage 1-1:1.0
[  247.773322] scsi 0:0:0:0: Direct-Access     SABRENT  ASM225           0    PQ: 0 ANSI: 6
[  247.774786] sd 0:0:0:0: [sda] 1875385008 512-byte logical blocks: (960 GB/894 GiB)
[  247.780964] sd 0:0:0:0: [sda] Write Protect is off
[  247.787939] sd 0:0:0:0: [sda] Mode Sense: 43 00 00 00
[  247.788431] sd 0:0:0:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[  247.802895]  sda: sda1
[  247.803136] sd 0:0:0:0: [sda] Attached SCSI disk
```
Full log is here: https://microbin.fenriv.co.ua:444/raw/sloth-bird

Comment (Ansuel) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:422]: Really interesting... Can you contact you later for further testing about this? Thanks a lot for the tests.

**Comment by Ansuel** [comment]
I guess the only blocker for this is the tool in fw utils

File: target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts (hunk 2 above)
Comment (SergeyFilippov) [reply target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-e8350-v1.dts:422]: > Really interesting... Can you contact you later for further testing about this? Thanks a lot for the tests.

Yes, sure, you welcome 😊. 
I will mark then this note as resolved.

**Comment by SergeyFilippov** [comment]
@Ansuel Sorry for a silly question, but i see the Makefile changes were done here `package/utils/firmware-utils/Makefile`, but not here `tools/firmware-utils/Makefile`. Is this supposed to be this way? [[commit]](https://github.com/openwrt/openwrt/commit/2b6772c82c6a4a498a7e5427b39ead2d60a42762)
 Maybe I'm doing something wrong, but for me the package is not being compiled and as result the tool is no present in `bin` during the image generation.

**Comment by Ansuel** [comment]
@SergeyFilippov do you have some kind of security option enabled of some sort? I can't rebase this pr on top of main for some reason...

And yes i also had to update the tools firmware-utils.

**Comment by SergeyFilippov** [comment]
> @SergeyFilippov do you have some kind of security option enabled of some sort? I can't rebase this pr on top of main for some reason...
> 
> And yes i also had to update the tools firmware-utils.

@Ansuel no security checks that I'm aware of and I was able to rebase the branch to main just now. The `allow edits for maintainers` is also set.

**Comment by SergeyFilippov** [comment]
@Ansuel Hi there, is there anything needed from my side about this changes?
If the issue with merging persist, I can re-create a merge request or anything required to fix it.

**Comment by Ansuel** [comment]
@SergeyFilippov yes it's strange. Can you please recreate the PR?

**Comment by SergeyFilippov** [comment]
> @SergeyFilippov yes it's strange. Can you please recreate the PR?

Of course. 
Closing this PR to create a new one.

**Pull request closed without merging**